plutus-tx-plugin/src/PlutusTx/Compiler/Types.hs

```diff
@@ -174,13 +177,69 @@ stableModuleCmp m1 m2 =
     -- See Note [Stable name comparisons]
     (GHC.moduleUnit m1 `GHC.stableUnitCmp` GHC.moduleUnit m2)
 
+class Monad m => MonadCoreM m where
```

Comment:
You forgot the`lookupThing` method.
Wouldn't it be better to add some orphan instances of the `MonadThings` class? Then only the `thNameToGhcName` method would be missing, but then you could create a subclass:

```
class MonadThings m => MonadCoreM m where
    thNameToGhcName = ...
```

https://hackage.haskell.org/package/ghc-9.2.7/docs/GHC-Types-TyThing.html#t:MonadThings